src/main/kotlin/cacophony/controlflow/functions/FunctionCallHandlerImpl.kt

```diff
@@ -0,0 +1,32 @@
+package cacophony.controlflow.functions
+
+import cacophony.controlflow.CFGNode
+import cacophony.controlflow.CFGNode.RegisterUse
+import cacophony.controlflow.HardwareRegister
+import cacophony.controlflow.Register
+import cacophony.semantic.syntaxtree.Definition.FunctionDeclaration
+
+open class FunctionCallHandlerImpl(
```

Comment:
If I understand, we'll have another implementation like _ForeignFunctionCallHandlerImpl_, so maybe _FunctionCallHandlerImpl_ can get a better name that shows that it's user created function? But I'm not sure, just a suggestion.

Reply:
I was thinking there was one FunctionCallHandlerImpl which just checks inside which function it uses. Maybe let's leave it up to decide for the person who will be implementing this?